File: Dairyncia-dotnet/Models/Address.cs
namespace Dairyncia.Models {
using System.ComponentModel.DataAnnotations;

    public class Address : BaseEntity
    {
        [Required(ErrorMessage ="Village name is required" )]
        [MaxLength(100, ErrorMessage ="Village name can not exceeds 100 characters")]
        [RegularExpression(@"^[a-zA-Z\s]+$", ErrorMessage = "Village name can contain only letters and spaces.")]
        public string Village { get; set; }

        [Required(ErrorMessage = "City name is required")]
        [MaxLength(100, ErrorMessage = "City name can not exceeds 100 characters")]
        [RegularExpression(@"^[a-zA-Z\s]+$", ErrorMessage = "City name can contain only letters and spaces.")]
        public string City { get; set; }

        [Required(ErrorMessage = "State name is required")]
        [MaxLength(100, ErrorMessage = "State name can not exceeds 100 characters")]
        [RegularExpression(@"^[a-zA-Z\s]+$", ErrorMessage = "State name can contain only letters and spaces.")]
        public string State { get; set; }

        [Required(ErrorMessage = "Pincode is required.")]
        [RegularExpression(@"^[1-9][0-9]{5}$", ErrorMessage = "Pincode must be a valid 6-digit Indian pincode.")]
        public string Pincode { get; set; }
    }
}

File: Dairyncia-Spring/src/main/java/com/dairyncia/entities/Address.java
package com.dairyncia.entities;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity
@Table(name="addresses")
@NoArgsConstructor
@Getter
@Setter
@ToString(callSuper=true)
public class Address extends BaseEntity{
	@NotBlank(message="Village name is required")
	@Size(max=50,message="Village name cannot exceed 50 characters")
	@Pattern(regexp= "^[a-zA-Z\\s]+$", message="Village name only contains letters and spaces")
	@Column(length=50,nullable=false)
	private String village;
	
	@NotBlank(message="City name is required")
	@Size(max=50,message="City name cannot exceed 50 characters")
	@Pattern(regexp= "^[a-zA-Z\\s]+$", message="City name only contains letters and spaces")
	@Column(length=50,nullable=false)
	private String city;
	
	@NotBlank(message="State name is required")
	@Size(max=50,message="State name cannot exceed 50 characters")
	@Pattern(regexp= "^[a-zA-Z\\s]+$", message="State name only contains letters and spaces")
	@Column(length=50,nullable=false)
	private String state;
	
	@NotBlank(message="Pincode is required")
	@Pattern(regexp= "^[1-9][0-9]{5}$", message="Pincode must be a valid 6-digit Indian pincode")
	@Column(length=6,nullable=false)
	private String pincode;
}
